File: app/src/main/java/com/kotlin/inaction/Student.java
package com.kotlin.inaction;

public class Student {
    private final String name;
    private int age;

    public Student(String _name, int _age) {
        name = _name;
        age = _age;
    }

    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }
}

File: app/src/main/java/com/kotlin/inaction/reference/classes_and_objects/02_properties_and_fields/Student.kt
package com.kotlin.inaction.reference.classes_and_objects.`02_properties_and_fields`

class Student(_name: String, _age: Int) {
    val name = _name
    var age = _age
    private var _grades: Map<String, Int>? = null
    val grades: Map<String, Int>
        get() {
            if (_grades == null) {
                _grades = loadGrades()
            }
            return _grades!!
        }

    private fun loadGrades(): MutableMap<String, Int> {
        println("loadGrades() called")
        val result = mutableMapOf<String, Int>()
        result.put("Chinese", 100)
        result.put("Math", 99)
        result.put("English", 100)
        return result
    }
}

fun main(args: Array<String>) {
    val student = Student("willwaywang6", 16)
    println("grades=${student.grades}")
    println("grades=${student.grades}")
}